plugins/user-authenticators/ldap/src/org/apache/cloudstack/ldap/LdapContextFactory.java

```diff
@@ -80,9 +79,13 @@ private void enableSSL(final Hashtable<String, String> environment) {
 
     private Hashtable<String, String> getEnvironment(final String principal, final String password, final String providerUrl, final boolean isSystemContext, Long domainId) {
         final String factory = _ldapConfiguration.getFactory();
-        final String url = providerUrl == null ? _ldapConfiguration.getProviderUrl(domainId) : providerUrl;
+        String url = providerUrl == null ? _ldapConfiguration.getProviderUrl(domainId) : providerUrl;
+        if (StringUtils.isEmpty(url) && domainId != null) {
```

Comment:
Can we do a `Strings.isNullOrEmpty()` instead?

Reply:
we can but getProviderUrl(domainId) returns "" at least